### PR TITLE
feat: Create; add candidate, candidate table, candidate detail, edit …

### LINKED ISSUE #137: [CM] Add Candidates: Name Entry
### Name entry with second names.

It does not allow first names with second names (e.g., Marie Jeremie). An exception should be made for this.

image

### LINKED ISSUE #138: [CM] Add Candidates: Position Selection
### Selection of position
An error message appears even after selecting a position.

image

### LINKED ISSUE #139: [CM] Add Candidates: Block Selection
### Block Section
An error message appears even after selecting a block section.

image

### LINKED ISSUE #140: [CM] Manage Candidates: Page Cannot Be Accessed
### Manage Candidates
The page manage-candidates is not functioning properly. See attachment below:

https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/3f9097c2-d00f-4bb1-b66b-2d5470ebb21d

### LINKED ISSUE #141: Candidates Management Module Issues
### Issues to be Addressed:

**Add Candidate**
- [x] #137
- [x] #138
- [x] #139

**Manage Candidate**
- [x] #140

### PR DESCRIPTION
**Pull request by hatdogguldo**
fixed #137 :    Change the regex validInput

fixed #138 :  Remove error message to the select fields in add candidate form
        
fixed #139:  Remove error message to the select fields in add candidate form

fixed #140 : [I created a php script for adding new column named "candidate_creation"](https://drive.google.com/drive/folders/1tKJrlJ9c2YdfNwNNtgLMUgcDOk-FRHtz?fbclid=IwAR2OB3lpLjHFirKu7OI_LAEu1Jt53dfxtAdG2032GfDB0lROGn8hSms1uiw)

close #141

### CODE REVIEW

File: src/images/candidate-profile/BPF.drawio.png
Comment (biellamariscotes) [inline src/images/candidate-profile/BPF.drawio.png:1]: Kindly remove the test data.

File: src/images/candidate-profile/BPF_Final.drawio.png
Comment (biellamariscotes) [inline src/images/candidate-profile/BPF_Final.drawio.png:1]: Kindly remove the test data.

File: src/images/candidate-profile/Proposed Business Flow Diagram_page-0001.png
Comment (biellamariscotes) [inline src/images/candidate-profile/Proposed Business Flow Diagram_page-0001.png:1]: Kindly remove the test data.

File: src/images/candidate-profile/Proposed ERD_page-0001.jpg
Comment (biellamariscotes) [inline src/images/candidate-profile/Proposed ERD_page-0001.jpg:1]: Kindly remove the test data.

File: src/images/candidate-profile/output-0000.jpg
Comment (biellamariscotes) [inline src/images/candidate-profile/output-0000.jpg:1]: Kindly remove the test data.

File: src/images/candidate-profile/sad_group3_2024-05-25_07.39pm.png
Comment (biellamariscotes) [inline src/images/candidate-profile/sad_group3_2024-05-25_07.39pm.png:1]: Kindly remove the test data.

File: src/manage-candidate.php [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+									</div>
+								</div>
+							</div>
+						</div>
+					</div>
+
+
+					<div class="container">
+						<div class="row justify-content-center">
+							<!-- VERIFIED TABLE -->
```
Comment (biellamariscotes) [inline src/manage-candidate.php]: Maintain proper documentation of comments. Replace as "CANDIDATES TABLE".

File: src/submission_handlers/update-candidate.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        $party_list = $_POST['party_list'];
+        $position_id = $_POST['position_id'];
+        $section_year = explode('-', $_POST['section']);
+        $year_level = $section_year[0];
+        $section = $section_year[1];
+
+        // Handle file upload
+        $photo_url = '';
+        if (isset($_FILES['photo']['name'])) {
+            $target_dir = __DIR__ . '/../images/candidate-profile/';
```
Comment (biellamariscotes) [inline src/submission_handlers/update-candidate.php]: The file path for the images of the candidates is incorrect. These images should be stored in the following directory: `src/user_data/orgname/candidate_imgs`.

**Review by biellamariscotes — CHANGES_REQUESTED** [review]

**Comment by biellamariscotes** [comment]
### ❗ Manage Candidates
- The page seems to be not showing me anything. Take a look:


https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/3739ed5d-b600-403e-9d14-15ac7717fa2a

<hr>


### Add Candidates - Issues Identified

- Name Entry: It does not allow me to enter first names with second names (e.g., Marie Jeremie). An exception should be made for this.
- Position Selection: An error message appears even after selecting a position.
- Block Section: An error message appears even after selecting a block section.


https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/90e18083-bd26-4596-b663-f079569cd839

Screenshot 2024-06-02 170249

These are critical bugs as they prevent the submission, so I cannot completely simulate the submission of the form.

I've created an issue ticket for this. Please refer to the linked issue in this PR.

<hr>

Prioritize addressing the bugs, particularly the broken page in Manage Candidates. Once that's resolved, I'll conduct a more thorough review and provide additional feedback on any critical issues if both of Candidate pages are functional. Thank you for your attention to this matter.

### Notes:
- Please avoid pushing test data (like the candidate images in your PR) in the commits.
- The excessive use of `!important` in your styles is concerning as it may override other styles unintentionally. Please use `!important` more sparingly to prevent this.

**Comment by hatdogguldo** [comment]
> ### ❗ Manage Candidates
> * The page seems to be not showing me anything. Take a look:

I created alter-table for candidate to add new key "candidate_creation".  
[Link](https://drive.google.com/drive/folders/1Ke1F_vxh0vj1zzWkpMjmeKA96uUizTsT?fbclid=IwAR2OB3lpLjHFirKu7OI_LAEu1Jt53dfxtAdG2032GfDB0lROGn8hSms1uiw)

File: src/submission_handlers/update-candidate.php (hunk 2 above)
Comment (hatdogguldo) [reply src/submission_handlers/update-candidate.php]: This part is already fixed.

Now this is my target_dir:
                                       '/../user_data/' . $org_name . '/candidate_imgs/'

**Comment by biellamariscotes** [comment]
<h2> Issues Identified</h2>

### Add Candidate
- Election Year is not being inserted into the database.

https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/73e88700-356a-4846-8e24-c00e4dd9b57a

### Manage Candidate
- Although I can see the codes in your files, the functionalities (sort, filter, search, pagination) are not working properly.

https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/4b062bd9-30de-431f-a1e2-0482fefc7f3f

<hr>

### ❗ Need to be addressed
- Remove the images on src/images/candidate-profile.
- Add the fixed issues (137, 138, 139, 140, 141) on this PR's description so that the included issues on this PR can be automatically done. Refer to this [PR's description ](https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/pull/117)on how to put the fixes.

<hr>

### Additional Notes
- In your **submission_handlers** files, you can use the existing classes in `query-handler.php` (located in `includes/classes`) to fetch data from the database.
- Make sure to also add the codes for the loader in your pages. Refer [here](https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/pull/156).

<hr>

After the **Needs to be addressed** have been done, we can merge this. You can address the identified issues, as well as other enhancements in additional notes, on your future PRs. I will just ticket issues to track these bugs.

Most of the problems are inconsistency with the UI, validations in the form, functionalities of the table— so kindly improve them.

**Comment by biellamariscotes** [comment]
@hatdogguldo Just following up on the requested changes. Please address them as soon as possible. Thanks!

**Comment by hatdogguldo** [comment]
fixed #137 :    Change the regex validInput

fixed #138 :  Remove error message to the select fields in add candidate form
        
fixed #139:  Remove error message to the select fields in add candidate form

fixed #140 : [I created a php script for adding new column named "candidate_creation"](https://drive.google.com/drive/folders/1tKJrlJ9c2YdfNwNNtgLMUgcDOk-FRHtz?fbclid=IwAR2OB3lpLjHFirKu7OI_LAEu1Jt53dfxtAdG2032GfDB0lROGn8hSms1uiw)

close #141

**Review by biellamariscotes — APPROVED** [review]

**Pull request merged**